remote-dependency-templates/README.md

```diff
@@ -3,9 +3,73 @@
   - Populate required `template-config` fields (defined in the README for each type)
 - List any dependency-specific service data under `dependency-data.<id>` in a service's `service.yml`
 
-## Terraform variables //TODO
+# Building dependency templates
+- Create a directory with a unique name. This will be the dependency's `type`
 
-## Notes
-The following are fields that exosphere passes to every dependency:
-- `app-name`: Name of the application as defined in `application.yml`
+## requirements.yml
+`requirements.yml` contains a single field: `required-fields`, which is a list of strings that define the fields that a user is required to populate
+under the `template-config` section of a dependency configuration.
 
+Example:
+```yml
+# requirements.yml
+
+required-fields:
+  - version
+```
+```yml
+# application.yml
+
+<dependency-id>:
+  type: <type>
+  template-config:
+    version: 0.0.1
+```
+
+## dependency.tf
+Add a terraform file, `dependency.tf`, which contains the public interfacing terraform modules for a dependency. The `dependency.tf` file should be built
+as a [Mustache template](https://github.com/hoisie/mustache). Any configuration information listed as a required field in `requirements.yml` (see above)
+will be rendered into `dependency.tf` using Mustache. The mustache variables must match those listed in `requirements.yml`, and hence in a user's `template-config`.
+Example:
+```
```

Comment:
Please add the syntax for all code blocks

Reply:
I'm not sure what the syntax block for `.tf` would be?

Reply:
Its `tf`. I added it